File: lab1_2_12/4.cpp
#include<bits/stdc++.h>
using namespace std;

void bfs(vector<int> adj[], int src, vector<int>&vis, vector<int>& res)
{
    vis[src] = 1;
    queue<int> q;
    q.push(src);

    while(!q.empty())
    {
        int node = q.front();
        q.pop();
        res.push_back(node);

        for(int ngb : adj[node])
        {
            if(!vis[ngb])
            {
                vis[ngb] = 1;
                q.push(ngb);
            }
        }
    }
}

int main()
{
    cout << "Enter no of vertices & edges : ";
    int n,m;
    cin >> n >> m;
    cout << "Enter each edge's src and dst node\n";
    vector<int> adj[n+1];
    for(int i=0;i<m;i++)
    {
        int x,y;
        cin >> x >> y;
        adj[x].push_back(y);
        adj[y].push_back(x);
    }

    vector<int> vis(n+1, 0);
    vector<int> res;
    bfs(adj, 1, vis, res);

    for(int i= 0;i<res.size();i++)
        cout << res[i] << " "; 
    cout << "\n";
    return 0;
}

File: lab1_2_12/4.py
import sys
import collections

def bfs(adj, src, vis, res):
    vis[src] = 1
    queue = collections.deque([src])

    while queue:
        node = queue.popleft()
        res.append(node)

        if node in adj:
            for ngb in adj[node]:
                if not vis[ngb]:
                    vis[ngb] = 1
                    queue.append(ngb)

def main():
    try:
        input_data = sys.stdin.read().split()
        if not input_data: return
        it = iter(input_data)
        
        n = int(next(it))
        m = int(next(it))
        
        adj = {}
        for _ in range(m):
            x = int(next(it))
            y = int(next(it))
            if x not in adj: adj[x] = []
            if y not in adj: adj[y] = []
            adj[x].append(y)
            adj[y].append(x)

        vis = [0] * (n + 1)
        res = []
        
        # Original code starts BFS from node 1
        bfs(adj, 1, vis, res)

        print(*(res))
    except (StopIteration, ValueError, EOFError):
        pass

if __name__ == "__main__":
    main()
